### PR TITLE
[fix] Change UEID as per RATS specification

### LINKED ISSUE #906: UEID extension doesn't comply with IETF RATS specification
https://www.ietf.org/archive/id/draft-ietf-rats-eat-21.html#section-4.2.1.1

First byte must be an type identifier. Then next bytes are the UEID itself.

To fix this, need to:
* expand the Manufacturer Serial Number field in IDEVID_CERT_ATTR fuse to be 128-bits
* Add a UEID type byte in IDEVID_CERT_ATTR

This means changes to ROM code and the certificate templates.

### PR DESCRIPTION
**Pull request by mhatrevi**
This fix updates the format of the UEID; it is now extended to 128-bits. Additionally, UEID type field is added to indicate the type of the UEID.

https://www.ietf.org/archive/id/draft-ietf-rats-eat-21.html#section-4.2.1.1

### CODE REVIEW

File: sw-emulator/app/src/main.rs [hunk 1]
```diff
@@ -350,9 +350,12 @@ fn main() -> io::Result<()> {
         // DWORD 00 - Flags
         cert[0] = flags.get();
         // DWORD 01 - 05 - IDEVID Subject Key Identifier (all zeroes)
-        // DWORD 06 - 07 - UEID / Manufacturer Serial Number
+        // DWORD 06 - 09 - UEID / Manufacturer Serial Number
         cert[6] = *args_ueid as u32;
         cert[7] = (*args_ueid >> 32) as u32;
+        cert[8] = (*args_ueid >> 64) as u32;
+        cert[9] = (*args_ueid >> 96) as u32;
+        cert[10] = 1; // UEID Type
```
Comment (varuns-nvidia) [inline sw-emulator/app/src/main.rs:358]: this should be cert[6] right?

Comment (mhatrevi) [reply sw-emulator/app/src/main.rs:358]: I defined UEID type as DWORD 10 in the fuse_idevid_cert_attr array:

    ManufacturerSerialNumber1 = 6,
    ManufacturerSerialNumber2 = 7,
    ManufacturerSerialNumber3 = 8,
    ManufacturerSerialNumber4 = 9,
    UeidType = 10,

I can make it DWORD 6 and move the UEID from DWORDs 7-10

Comment (varuns-nvidia) [reply sw-emulator/app/src/main.rs:358]: no need. was just confused whether this array was indexing cert blocks or fuse blocks.

Comment (varuns-nvidia) [reply sw-emulator/app/src/main.rs]: I see you did it anyway, that works.

Comment (mhatrevi) [reply sw-emulator/app/src/main.rs]: Updated fuse_idevid_cert_attr fields to make this more obvious:

    UeidType = 6,
    ManufacturerSerialNumber1 = 7,
    ManufacturerSerialNumber2 = 8,
    ManufacturerSerialNumber3 = 9,
    ManufacturerSerialNumber4 = 10,

**Comment by varuns-nvidia** [comment]
UEID is used in FMC Alias and RT Alias as well so that collateral needs updating.

I will also update https://github.com/chipsalliance/Caliptra/pull/132/files to note the UEID encoding change.

**Review by jhand2 — DISMISSED** [review]

**Review by varuns-nvidia — DISMISSED** [review]

**Comment by korran** [comment]
Smoke-test DER diffs:

```patch
$ diff --color --unified <(git show HEAD^:test/tests/smoke_testdata/idevid_csr.der | openssl asn1parse -inform der | cut -c 6-) <(git show HEAD:test/tests/smoke_testdata/idevid_csr.der | openssl asn1parse -inform der  | cut -c 6-)
--- /dev/fd/63	2023-10-04 22:57:27.295731105 -0700
+++ /dev/fd/62	2023-10-04 22:57:27.295731105 -0700
@@ -1,5 +1,5 @@
-:d=0  hl=4 l= 428 cons: SEQUENCE          
-:d=1  hl=4 l= 305 cons: SEQUENCE          
+:d=0  hl=4 l= 436 cons: SEQUENCE          
+:d=1  hl=4 l= 314 cons: SEQUENCE          
 :d=2  hl=2 l=   1 prim: INTEGER           :00
 :d=2  hl=2 l= 101 cons: SEQUENCE          
 :d=3  hl=2 l=  24 cons: SET               
@@ -15,11 +15,11 @@
 :d=4  hl=2 l=   7 prim: OBJECT            :id-ecPublicKey
 :d=4  hl=2 l=   5 prim: OBJECT            :secp384r1
 :d=3  hl=2 l=  98 prim: BIT STRING        
-:d=2  hl=2 l=  77 cons: cont [ 0 ]        
-:d=3  hl=2 l=  75 cons: SEQUENCE          
+:d=2  hl=2 l=  86 cons: cont [ 0 ]        
+:d=3  hl=2 l=  84 cons: SEQUENCE          
 :d=4  hl=2 l=   9 prim: OBJECT            :Extension Request
-:d=4  hl=2 l=  62 cons: SET               
-:d=5  hl=2 l=  60 cons: SEQUENCE          
+:d=4  hl=2 l=  71 cons: SET               
+:d=5  hl=2 l=  69 cons: SEQUENCE          
 :d=6  hl=2 l=  18 cons: SEQUENCE          
 :d=7  hl=2 l=   3 prim: OBJECT            :X509v3 Basic Constraints
 :d=7  hl=2 l=   1 prim: BOOLEAN           :255
@@ -28,9 +28,9 @@
 :d=7  hl=2 l=   3 prim: OBJECT            :X509v3 Key Usage
 :d=7  hl=2 l=   1 prim: BOOLEAN           :255
 :d=7  hl=2 l=   4 prim: OCTET STRING      [HEX DUMP]:03020204
-:d=6  hl=2 l=  22 cons: SEQUENCE          
+:d=6  hl=2 l=  31 cons: SEQUENCE          
 :d=7  hl=2 l=   6 prim: OBJECT            :2.23.133.5.4.4
-:d=7  hl=2 l=  12 prim: OCTET STRING      [HEX DUMP]:300A04080000000000000000
+:d=7  hl=2 l=  21 prim: OCTET STRING      [HEX DUMP]:301304110000000000000000000000000000000000
 :d=1  hl=2 l=  10 cons: SEQUENCE          
 :d=2  hl=2 l=   8 prim: OBJECT            :ecdsa-with-SHA384
-:d=1  hl=2 l= 105 prim: BIT STRING        
+:d=1  hl=2 l= 104 prim: BIT STRING        
```

```patch
$ diff --color --unified <(git show HEAD^:test/tests/smoke_testdata/ldevid_cert.der | openssl asn1parse -inform der | cut -c 6-) <(git show HEAD:test/tests/smoke_testdata/ldevid_cert.der | openssl asn1parse -inform der  | cut -c 6-)
--- /dev/fd/63	2023-10-04 22:58:31.715066435 -0700
+++ /dev/fd/62	2023-10-04 22:58:31.715066435 -0700
@@ -1,5 +1,5 @@
-:d=0  hl=4 l= 651 cons: SEQUENCE          
-:d=1  hl=4 l= 529 cons: SEQUENCE          
+:d=0  hl=4 l= 662 cons: SEQUENCE          
+:d=1  hl=4 l= 540 cons: SEQUENCE          
 :d=2  hl=2 l=   3 cons: cont [ 0 ]        
 :d=3  hl=2 l=   1 prim: INTEGER           :02
 :d=2  hl=2 l=  20 prim: INTEGER           :25EEEF9A4C61D4B9E3D94BEA46F9A12AC6887CE2
@@ -31,8 +31,8 @@
 :d=4  hl=2 l=   7 prim: OBJECT            :id-ecPublicKey
 :d=4  hl=2 l=   5 prim: OBJECT            :secp384r1
 :d=3  hl=2 l=  98 prim: BIT STRING        
-:d=2  hl=2 l= 126 cons: cont [ 3 ]        
-:d=3  hl=2 l= 124 cons: SEQUENCE          
+:d=2  hl=3 l= 136 cons: cont [ 3 ]        
+:d=3  hl=3 l= 133 cons: SEQUENCE          
 :d=4  hl=2 l=  18 cons: SEQUENCE          
 :d=5  hl=2 l=   3 prim: OBJECT            :X509v3 Basic Constraints
 :d=5  hl=2 l=   1 prim: BOOLEAN           :255
@@ -41,9 +41,9 @@
 :d=5  hl=2 l=   3 prim: OBJECT            :X509v3 Key Usage
 :d=5  hl=2 l=   1 prim: BOOLEAN           :255
 :d=5  hl=2 l=   4 prim: OCTET STRING      [HEX DUMP]:03020204
-:d=4  hl=2 l=  22 cons: SEQUENCE          
+:d=4  hl=2 l=  31 cons: SEQUENCE          
 :d=5  hl=2 l=   6 prim: OBJECT            :2.23.133.5.4.4
-:d=5  hl=2 l=  12 prim: OCTET STRING      [HEX DUMP]:300A04080000000000000000
+:d=5  hl=2 l=  21 prim: OCTET STRING      [HEX DUMP]:301304110000000000000000000000000000000000
 :d=4  hl=2 l=  29 cons: SEQUENCE          
 :d=5  hl=2 l=   3 prim: OBJECT            :X509v3 Subject Key Identifier
 :d=5  hl=2 l=  22 prim: OCTET STRING      [HEX DUMP]:041421EEEF9A4C61D4B9E3D94BEA46F9A12AC6887CE2
```

**Review by korran — APPROVED** [review]

**Review by jhand2 — APPROVED** [review]

**Review by bluegate010 — APPROVED** [review]

File: fmc/tests/test_rtalias.rs [hunk 2]
```diff
@@ -82,8 +82,8 @@ fn test_fht_info() {
 
     let data = hw.mailbox_execute(TEST_CMD_READ_FHT, &[]).unwrap().unwrap();
     let fht = FirmwareHandoffTable::read_from_prefix(data.as_bytes()).unwrap();
-    assert_eq!(fht.ldevid_tbs_size, 533);
-    assert_eq!(fht.fmcalias_tbs_size, 769);
+    assert_eq!(fht.ldevid_tbs_size, 544);
```
Comment (FerralCoder) [inline fmc/tests/test_rtalias.rs:85]: didn't the size only increase by 9 rather than 11?

Comment (mhatrevi) [reply fmc/tests/test_rtalias.rs:85]: Alignment added 2 bytes.

Comment (FerralCoder) [reply fmc/tests/test_rtalias.rs:85]: got it

**Review by FerralCoder — APPROVED** [review]

**Pull request merged**